WordPressShared/WordPressShared/Core/Views/WPStyleGuide+DynamicType.swift

```diff
@@ -0,0 +1,182 @@
+/// Extension on WPStyleGuide to use Dynamic Type fonts.
+///
+extension WPStyleGuide {
+    static let defaultTableViewRowHeight: CGFloat = 44.0
+
+    static let notoLoaded = { () -> Bool in
+        WPFontManager.loadNotoFontFamily()
+        return true
+    }()
+
+    /// Configures a table to automatically resize its rows according to their content.
+    ///
+    /// - Parameters:
+    ///     - tableView: The tableView to configure.
+    ///
+    public class func configureAutomaticHeightRowsForTableView(_ tableView: UITableView) {
+        tableView.rowHeight = UITableViewAutomaticDimension
+        tableView.estimatedRowHeight = defaultTableViewRowHeight
+    }
+
+    /// Configures a label with the default system font with the specified style.
+    ///
+    /// - Parameters:
+    ///     - label: The label to configure.
+    ///     - style: The desired UIFontTextStyle.
+    ///
+    public class func configureLabel(_ label: UILabel, forTextStyle style: UIFontTextStyle) {
+        label.font = UIFont.preferredFont(forTextStyle: style)
+        label.adjustsFontForContentSizeCategory = true
+    }
+
+    /// Configures a label with the default system font with the specified style and traits.
+    ///
+    /// - Parameters:
+    ///     - label: The label to configure.
+    ///     - style: The desired UIFontTextStyle.
+    ///     - traits: The desired UIFontDescriptorSymbolicTraits.
+    ///
+    public class func configureLabel(_ label: UILabel, forTextStyle style: UIFontTextStyle, withTraits traits: UIFontDescriptorSymbolicTraits) {
+        label.font = self.fontForTextStyle(style, withTraits: traits)
+        label.adjustsFontForContentSizeCategory = true
+    }
+
+    /// Configures a label with the default system font with the specified style and weight.
+    ///
+    /// - Parameters:
+    ///     - label: The label to configure.
+    ///     - style: The desired UIFontTextStyle.
+    ///     - weight: The desired weight (UIFontWeightUltraLight, UIFontWeightThin, UIFontWeightLight,
+    ///       UIFontWeightRegular, UIFontWeightMedium, UIFontWeightSemibold, UIFontWeightBold,
+    ///       UIFontWeightHeavy, UIFontWeightBlack).
+    ///
+    public class func configureLabel(_ label: UILabel, forTextStyle style: UIFontTextStyle, withWeight weight: CGFloat) {
```

Comment:
Yes, I'm repeating myself. `withWeight` to just `weight`?

I'll stop now. You get what I'm saying ;)